components/movementsensor/merged/merged.go

```diff
@@ -0,0 +1,376 @@
+}
+
+func (m *merged) Accuracy(ctx context.Context, extra map[string]interface{}) (map[string]float32, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	accMap := make(map[string]float32)
+	var errs error
+
+	if m.ori != nil {
+		oriAcc, err := m.ori.Accuracy(ctx, extra)
+		if err != nil {
+			// replace entire map with a map that shows that it has errors
+			oriAcc = map[string]float32{
+				m.ori.Name().ShortName() + errStrAccuracy: float32(math.NaN()),
+			}
+			errs = multierr.Combine(errs, err)
+		}
+		maps.Copy(accMap, mapWithSensorName(m.ori.Name().ShortName(), oriAcc))
+	}
+
+	if m.pos != nil {
+		posAcc, err := m.pos.Accuracy(ctx, extra)
+		if err != nil {
+			posAcc = map[string]float32{
+				m.pos.Name().ShortName() + errStrAccuracy: float32(math.NaN()),
+			}
+			errs = multierr.Combine(errs, err)
+		}
+		maps.Copy(accMap, mapWithSensorName(m.pos.Name().ShortName(), posAcc))
+	}
+
+	if m.compass != nil {
+		compassAcc, err := m.compass.Accuracy(ctx, extra)
+		if err != nil {
+			compassAcc = map[string]float32{
+				m.compass.Name().ShortName() + errStrAccuracy: float32(math.NaN()),
+			}
+			errs = multierr.Combine(errs, err)
+		}
+		maps.Copy(accMap, mapWithSensorName(m.compass.Name().ShortName(), compassAcc))
+	}
+
+	if m.linVel != nil {
+		linvelAcc, err := m.linVel.Accuracy(ctx, extra)
+		if err != nil {
+			linvelAcc = map[string]float32{
+				m.linVel.Name().ShortName() + errStrAccuracy: float32(math.NaN()),
+			}
+			errs = multierr.Combine(errs, err)
+		}
+		maps.Copy(accMap, mapWithSensorName(m.linVel.Name().ShortName(), linvelAcc))
+	}
+
+	if m.angVel != nil {
+		angvelAcc, err := m.angVel.Accuracy(ctx, extra)
+		if err != nil {
+			angvelAcc = map[string]float32{
+				m.angVel.Name().ShortName() + errStrAccuracy: float32(math.NaN()),
+			}
+			errs = multierr.Combine(errs, err)
+		}
+		maps.Copy(accMap, mapWithSensorName(m.angVel.Name().ShortName(), angvelAcc))
+	}
+
+	if m.linAcc != nil {
+		linAccAcc, err := m.linAcc.Accuracy(ctx, extra)
+		if err != nil {
+			linAccAcc = map[string]float32{
+				m.linAcc.Name().ShortName() + errStrAccuracy: float32(math.NaN()),
+			}
+			errs = multierr.Combine(errs, err)
+		}
+		maps.Copy(accMap, mapWithSensorName(m.linAcc.Name().ShortName(), linAccAcc))
+	}
+
+	return accMap, errs
+}
+
+func (m *merged) Properties(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	return &movementsensor.Properties{
+		OrientationSupported:        m.ori != nil,
+		PositionSupported:           m.pos != nil,
+		CompassHeadingSupported:     m.compass != nil,
+		LinearVelocitySupported:     m.linVel != nil,
+		AngularVelocitySupported:    m.angVel != nil,
+		LinearAccelerationSupported: m.linAcc != nil,
+	}, nil
+}
+
+func (m *merged) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
+	// we're already in lock in this driver
+	// don't lock the mutex again for the Readings call
```

Comment:
Eh... this comment is slightly misleading. We're not already in lock in this driver, but calling `movementsensor.Readings` is going to call a bunch of functions in this class that each lock the mutex and _that's_ why it's important not to lock it here.

Reply:
How about this, please make a suggestion on github  and I'll commit it in.